src/modules/roc_audio/resampler_writer.cpp

```diff
@@ -73,8 +75,13 @@ void ResamplerWriter::write(Frame& input) {
             resampler_.renew_buffers(window_[0], window_[1], window_[2]);
 
             Frame out_frame(output_.data(), output_.size());
+
             while (resampler_.resample_buff(out_frame)) {
                 writer_.write(out_frame);
+
+                if (poison_) {
```

Comment:
We should do poisoning when the new frame begins, before writing any samples to it. This `if (poison_)` point is exactly that moment: we have done with the previous output frame and are starting a new one using the same buffer (maybe immediately or maybe on next write() call). The very first frame will not be poisoned thought, but I thinks it's not a big problem.

Alternatively, we could detect the new frame start before calling resample_buff(), but this would require adding new conditions to write() method. I wanted to avoid adding clever logic specific just for poisoning. I think the implementation of poisoning should be as stupid as possible because most bugs in such a feature would be completely undetectable.

Reply:
Agree, if we don't poison the first frame it isn't a big deal but it seems odd.
We introduce some kind of contract that `EVERY` frame created by various
components will be poisoned. Now you want to make some components poison
every new frame while another components may skip the first frame and poison
all the rest.

Reply:
> We introduce some kind of contract that EVERY frame created by various components will be poisoned.

Well, we didn't state that anywhere :) What we did state is that this option poisons something somewhere which may help to detect uninitialized samples.

Reply:
Ok, I got it. Without `PoisonWriter` it wasn't clear for me, don't know why :)